main.cpp

```diff
@@ -91,15 +91,113 @@ good to go!
  Wait for my code review.
  */
 
+
 #include <iostream>
-int main() 
+#include <math.h> 
+
+struct FloatType
+{
+    float add(float, float);
+    float subtract(float, float);
+    float multiply(float, float);
+    float divide(float, float);
+};
+    
+float FloatType::add(float lhs, float rhs) 
+{
+     return lhs + rhs; 
+}
+
+float FloatType::subtract(float lhs, float rhs)
+{
+    return lhs - rhs;    
+}
+
+float FloatType::multiply(float lhs, float rhs)
+{
+    return lhs * rhs;
+}
+
+float FloatType::divide(float lhs, float rhs )
+{
+    if( rhs == 0.f )  
+    { 
+         std::cout << "warning, floating point division by zero returns 'inf' ! " << std::endl;
+        return lhs/rhs;
+    }
+    
+    return lhs / rhs;
+}
```

Comment:
FIXME:
indentation of std::cout statement
no need for extra `return` statement inside `if()` block
- [ ] revise